lib/chef/resource/chocolatey_installer.rb

```diff
@@ -0,0 +1,198 @@
+            action :uninstall
+          end
+          ```
+
+          **Install Chocolatey with Parameters**
+
+          ```ruby
+          chocolatey_installer 'latest' do
+            action :install
+            download_url "https://www.contoso.com/foo"
+            chocolatey_version '2.12.24'
+          end
+          ```
+
+          **Upgrade Chocolatey with Parameters**
+
+          ```ruby
+          chocolatey_installer 'latest' do
+            action :upgrade
+            chocolatey_version '2.12.24'
+          end
+          ```
+      DOC
+
+      allowed_actions :install, :uninstall, :upgrade
+
+      property :download_url, String,
+        description: "The URL to download Chocolatey from. This defaults to the value of $env:ChocolateyDownloadUrl, if it is set, and otherwise falls back to the official Chocolatey community repository to download the Chocolatey package. It can be used for offline installation by providing a path to a Chocolatey.nupkg."
+
+      property :chocolatey_version, String,
+        description: "Specifies a target version of Chocolatey to install. By default, the latest stable version is installed. This will use the value in $env:ChocolateyVersion by default, if that environment variable is present. This parameter is ignored if download_url is set."
+
+      property :use_native_unzip, [TrueClass, FalseClass], default: false,
+        description: "If set, uses built-in Windows decompression tools instead of 7zip when unpacking the downloaded nupkg. This will be set by default if use_native_unzip is set to a value other than 'false' or '0'. This parameter will be ignored in PS 5+ in favour of using the Expand-Archive built in PowerShell cmdlet directly."
+
+      property :ignore_proxy, [TrueClass, FalseClass], default: false,
+        description: "If set, ignores any configured proxy. This will override any proxy environment variables or parameters. This will be set by default if ignore_proxy is set to a value other than 'false' or '0'."
+
+      property :proxy_url, String,
+        description: "Specifies the proxy URL to use during the download."
+
+      property :proxy_user, String,
+        description: "The username to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_password are set"
+
+      property :proxy_password, String,
+        description: "The password to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_user are set"
+
+      load_current_value do
+        current_state = is_choco_installed?
+        current_value_does_not_exist! if current_state == false
+        current_state
+      end
+
+      def is_choco_installed?
+        ::File.exist?("#{ENV["ALLUSERSPROFILE"]}\\chocolatey\\bin\\choco.exe")
+      end
+
+      def get_choco_version
+        powershell_exec("choco --version").result
+      end
+
+      def existing_version
```

Comment:
Does it conform to gem versions? We have other version helpers if not. Gem versions are very strict.

Reply:
BTW, got a pointer for me? Everything I read says to use Gem::Version

Reply:
If we're just comparing two *specific* gem versions, `Gem::Version` sounds good to me. If we're looking to have more general version specifiers (`~>`, etc...), it might still work, but we might want to consider how the user would expect things to work in context.

Reply:
I'm a total dork, ignore me. I understand now. I was overthinking this.